qiskit/qpy/binary_io/schedules.py

```diff
@@ -103,19 +103,25 @@ def _read_discriminator(file_obj, version):
     return Discriminator(name=name, **params)
 
 
-def _loads_symbolic_expr(expr_bytes):
-    from sympy import parse_expr
-
+def _loads_symbolic_expr(expr_bytes, use_symengine=False):
     if expr_bytes == b"":
         return None
+    if use_symengine:
+        _optional.HAS_SYMENGINE.require_now("loads_symbolic_expr")
+        from symengine.lib.symengine_wrapper import (  # pylint: disable = no-name-in-module
+            load_basic,
+        )
 
-    expr_txt = zlib.decompress(expr_bytes).decode(common.ENCODE)
-    expr = parse_expr(expr_txt)
+        expr = load_basic(zlib.decompress(expr_bytes))
```

Comment:
Do we still need to zlib compress the symengine binary serialization? Maybe @nkanazawa1989 or @wshanks has some insight here, but I was thinking this was done for the sympy path because the string representation from sympy will get pretty verbose for some of the more involved pulse functions. But I'm not sure how large the symengine [cereal](https://uscilab.github.io/cereal/)  serialization is and whether spending cpu time on inline compression here is worth it.

Reply:
hmmm I think it is probably still worth it. I checked for the schedule [in the unit tests](https://github.com/Qiskit/qiskit/pull/10820/files#diff-596d0a657d9b71620a28d347b01b775af4fb8bac2e1fd435f269cbb2b169d8b3R425). And I get that the uncompressed string vs compressed string sizes are:

```
use_symengine=False: 161 bytes uncompressed -> 101 bytes compressed (dump = 2.8 s)
use_symengine=True: 429 bytes uncompressed -> 143 bytes compressed (dump = 0.00256 s)
```

and the dump operation is 0.00005s  slower with compression vs without.

Reply:
Heh, I guess I was worried about nothing since the data sizes we're talking about are sufficiently small zlib is sufficiently fast.